crates/core/src/prelude/src/memory-handle.ts

```diff
@@ -0,0 +1,74 @@
+declare global {
+  interface MemoryHandle {
+    offset: number;
+    len: number;
```

Comment:
This touches a forward-looking thing – _technically_ `offset` and `len` (and any other `I64` value) may be represented as `bigint` in the near-to-mid term. I'm on the fence as to whether it's worth changing these definitions to `number | bigint` _ahead_ of merging this, or if it's preferable to merge now and roll out the more permissive type later.

Reply:
I have no opinion on this, so just tell me what you guys decide and I'll adjust :)

In any case it may make sense to at least use the I64 type for these? Or PTR for the offsets?

Reply:
I think that would work; I'll take care of that after we land this PR!

Reply:
I might first get back to this during the weekend, but I'm down for doing it. If you want it faster though then go ahead :)

Reply:
Ok, once I am able to manually run through some tests, let's merge as-is, and then we can follow-up with any additional changes. I think this is great for now!